keps/sig-api-machinery/4355-coordinated-leader-election/README.md

```diff
@@ -440,6 +451,91 @@ set of candidates and selected strategy is the same as before.
+
+| Component | Binary Version | Emulation Version | Priority | Leader |
+|-----------|----------------|-------------------|----------|--------|
+| C1        | V2             | V2                | 100      | Y      |
+| C2        | V2             | V1                |          |        |
+| C3        | V2             | V1                |          |        |
+
+##### 4.1. Upgrade Completion
+After the upgrade is finished, all components are running Binary Version 2 and are emulating Version 2. C1 remains the leader due to its set priority.
+
+| Component | Binary Version | Emulation Version | Priority | Leader |
+|-----------|----------------|-------------------|----------|--------|
+| C1        | V2             | V2                | 100      | Y      |
+| C2        | V2             | V2                |          |        |
+| C3        | V2             | V2                |          |        |
+
+##### 4.2 Update rollback
+Should an issue arise with C1 requiring a rollback, we can unset its priority. This will enable CLE to select C2, which contains the oldest emulated version.
+
+| Component | Binary Version | Emulation Version | Priority | Leader |
+|-----------|----------------|-------------------|----------|--------|
+| C1        | V2 -> V1       | V2 -> V1          |          |        |
+| C2        | V2             | V1                |          | Y      |
+| C3        | V2             | V1                |          |        |
+
+##### 5. Priority Persistence
+Unless the cluster administrator resets the priority, C1 will always remain the leader. When a component gets upgraded or downgraded, it may create a new release candidate, causing the priority to reset.
+
+#### Consideration for Stale Priorities
+A concern with the priority field is the potential for "stale priorities" – a priority set temporarily and not subsequently cleared. This could prevent the Coordinated Leader Election (CLE) system from selecting a more appropriate leader.
+We considered exposing a Time-To-Live (TTL) for priority in the `LeaseCandidateSpec`, where the CLE system would ignore a priority once its TTL expired. While this directly addresses the "temporary" nature of many priority assignments, we've decided not to include it in this initial phase due to several complexities:
+- Implementation and Semantics: Defining the precise data type and behavior for a TTL (e.g., time.Duration vs. time.Time, resetting logic) adds significant complexity.
+- User Rationalization: Adding a third field (ttl) to an already multi-faceted leader election logic (strategy + priority) greatly increases the cognitive load for users to understand and manage leader selection effectively.
+
+Therefore, in this initial iteration, managing priority lifecycles will be an operational responsibility, requiring manual clearance or updates. We may revisit TTL or similar automated mechanisms in future iterations after gaining more experience with the priority field.
```

Comment:
This feels like an important but unsolved aspect of this design. Let's add a graduation criteria line item about this.